src/EditorFeatures/CSharp/AutomaticCompletion/AutomaticLineEnderCommandHandler.cs

```diff
@@ -273,18 +317,900 @@ private static IEnumerable<SyntaxNode> GetOwningNodes(SyntaxNode root, int posit
+                or ArrowExpressionClauseSyntax;
+
+        private static SyntaxNode? OwningNode(SyntaxNode n)
+            => n is ArrowExpressionClauseSyntax ? n.Parent : n;
+
+        #endregion
+
+        #region BraceModification
+
+        protected override void ModifySelectedNode(
+            AutomaticLineEnderCommandArgs args,
+            Document document,
+            SyntaxNode selectedNode,
+            int caretPosition,
+            CancellationToken cancellationToken)
+        {
+            var root = document.GetRequiredSyntaxRootSynchronously(cancellationToken);
+            // Add braces for the selected node
+            if (ShouldAddBraces(selectedNode, caretPosition))
+            {
+                // For these syntax node, braces pair could be easily added by modify the syntax tree
+                if (selectedNode is BaseTypeDeclarationSyntax
+                    or BaseMethodDeclarationSyntax
+                    or LocalFunctionStatementSyntax
+                    or FieldDeclarationSyntax
+                    or EventFieldDeclarationSyntax
+                    or AccessorDeclarationSyntax
+                    or ObjectCreationExpressionSyntax
+                    or WhileStatementSyntax
+                    or ForEachStatementSyntax
+                    or ForStatementSyntax
+                    or LockStatementSyntax
+                    or UsingStatementSyntax
+                    or DoStatementSyntax
+                    or IfStatementSyntax
+                    or ElseClauseSyntax)
+                {
+                    // Add the braces and get the next caretPosition
+                    var (newRoot, nextCaretPosition) = AddBraceToSelectedNode(document, root, selectedNode, cancellationToken);
+                    if (document.Project.Solution.Workspace.TryApplyChanges(document.WithSyntaxRoot(newRoot).Project.Solution))
+                    {
+                        args.TextView.TryMoveCaretToAndEnsureVisible(new SnapshotPoint(args.SubjectBuffer.CurrentSnapshot, nextCaretPosition));
+                    }
+
+                    return;
+                }
+                else
+                {
+                    // For the rest of the syntax node,
+                    // like try statement
+                    // class Bar
+                    // {
+                    //      void Main()
+                    //      {
+                    //          tr$$y
+                    //      }
+                    // }
+                    // In this case, the last close brace of 'void Main()' would be thought as a part of the try statement,
+                    // and the last close brace of 'Bar' would be thought as a part of Main()
+                    // So for these case, just find the missing open brace position and directly insert '()' to the document
+                    // 1. Find the position to insert braces.
+                    var insertionPosition = GetBraceInsertionPosition(selectedNode);
```

Comment:
Tag here for any reviewer interested in this part.
I am not sure if there is a better way to implement this since for (namespace, switch statement, try, catch, finally, indexer declaration) 
The problem is, if it doesn't have braces, the parser has a different logic to think the code,
```C#
class Bar
{
    void Sub() 
    {
          switch$$
    }
}
```
here parser think 'switch' has the '}' next to it, and its open brace is missing. But in fact it should be a brace for the method.

here the implantation  is just directly insert a brace pair at the missing open brace position. Any better idea to fix the syntax tree is welcome.